src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -143,86 +144,112 @@ case class WdlNamespaceWithWorkflow(importedAs: Option[String],
+  def load(wdlSource: WdlSource, importResolver:ImportResolver): WdlNamespace = {
+    load(wdlSource, "string",  Seq(importResolver), None)
+  }
+
+  def load(wdlSource: WdlSource, importResolver: Seq[ImportResolver]): WdlNamespace = {
     load(wdlSource, "string", importResolver, None)
   }
 
   def load(wdlSource: WdlSource, resource: String): WdlNamespace = {
-    load(wdlSource, resource, localImportResolver, None)
+    load(wdlSource, resource, Seq(fileResolver), None)
   }
 
   def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver): WdlNamespace = {
-    load(wdlSource, resource, importResolver, None)
+    load(wdlSource, resource, Seq(importResolver), None)
   }
 
-  private def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver, importedAs: Option[String]): WdlNamespace = {
+  private def load(wdlSource: WdlSource, resource: String, importResolver: Seq[ImportResolver], importedAs: Option[String]): WdlNamespace = {
     WdlNamespace(AstTools.getAst(wdlSource, resource), wdlSource, importResolver, importedAs, root = true)
   }
 
-  def apply(ast: Ast, source: WdlSource, importResolver: ImportResolver, namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
+  def apply(ast: Ast, source: WdlSource, importResolvers: Seq[ImportResolver], namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
     val imports = for {
       importAst <- Option(ast).map(_.getAttribute("imports")).toSeq
       importStatement <- importAst.astListAsVector.map(Import(_))
     } yield importStatement
 
+    def tryResolve(str: String): Option[WdlSource] = {
+      importResolvers map { resolver =>
+        Try(resolver(str))
+      } collectFirst {
+        case Success(s) => s
+        case Failure(error) => throw error
+      }
+    }
     /**
-      * For import statements with an 'as' clause, this translates to a sub-namespace
+      * Translates import statements to sub-namespaces
       */
     val namespaces: Seq[WdlNamespace] = for {
-      imp <- imports if imp.namespaceName.isDefined
-      source = importResolver(imp.uri)
-      if source.length > 0
-    } yield WdlNamespace.load(source, imp.uri, importResolver, imp.namespaceName)
+      imp <- imports
+      source = tryResolve(imp.uri)
+      if source.isDefined
+    } yield WdlNamespace.load(source.get, imp.uri, importResolvers,Option(imp.namespaceName.getOrElse(imp.uri)))
 
     /**
-      * For import statements without an 'as' clause, this is treated as a c-style #include
+      * For import statements without an 'as' clause, c# style include?
       */
-    val nonNamespacedImports = for {
+    val nonNamespacedImportsScopeAsts = for {
       imp <- imports if imp.namespaceName.isEmpty
-      source = importResolver(imp.uri)
-      ast = AstTools.getAst(source, imp.uri)
+      source = tryResolve(imp.uri)
+      if source.isDefined
+      ast = AstTools.getAst(source.get, imp.uri)
```

Comment:
`namespaces` above already creates a namespaces for all imports (with and without "as" clause).
Here imports without "as" clause are parsed a second time. It would be nice to find a way to avoid that.